raffle.js

```diff
@@ -1,40 +1,55 @@
 #!/usr/bin/env node
 
-var fs = require('fs');
-
-var name = 'raffle.json';
-var noop = function() {};
-
-function random(count){
-
-  read(function(err, data) {
-    var i = 0;
-    while(i < count){
-      var a = (Math.random()*600).toFixed(0);
-      if(!~data.indexOf(a)){
-        data.push(a);
-        console.log(a);
-        i++;
-      }
-    }
-    write(data, noop);
-  })
-}
-
-function write(data, cb) {
-  fs.writeFile(name, JSON.stringify(data), cb);
-}
-
-function read(cb) {
-  fs.readFile(name, function(err, data) {
-    if (err) {
-      if (err.code == 'ENOENT') return cb(null, ["0"]);
-      return cb(err);
+var fs    = require('fs');
+var path  = require('path')
+
+function Raffle(people){
+  if(!(this instanceof Raffle)){
+    return new Raffle(people);
+  }
+  this.people    = people;
+  this.blacklist  = [ 0 ];
+  this.dbFile     = path.basename(__dirname, '.js') + '.json';
+};
+
+Raffle.prototype.random = function(size){
+  this.read();
+  var i = 0;
+  var luckyBoys = [];
+  while(this.blacklist.length <= this.people &&  i < size){
+    var r = ((Math.random() * this.people).toFixed(0));
```

Comment:
Here the `r` is a string. but the `blacklist` seems a list of numbers. So the `indexOf` will be always `false`.